notesapplication/notes/api_v1/views.py

```diff
@@ -23,25 +21,73 @@ class NotesViewSet(viewsets.ModelViewSet):
     """
 
     queryset = Note.objects.all()
-    permission_classes = (IsAuthenticated,)
-    serializer_class = NotesSerializer
+    permission_classes = (IsAuthenticated, SuperUserReadOnly)
+    serializer_class = NoteCommentSerializer
+    filter_backends = [
+        filters.SearchFilter,
+    ]
+    search_fields = ["text"]
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
+            return NoteCommentSerializer
+        return NotesSerializer
 
     def get_queryset(self):
         queryset = Note.objects.all()
         user = self.request.user
         is_shared = self.request.query_params.get("is_shared")
         is_archive = self.request.query_params.get("is_archive")
 
-        if is_archive:
+        if is_archive == "True":
             queryset = Note.objects.all()
             user = self.request.user
             queryset = queryset.filter(user=user, archive=1)
             return queryset
         if is_shared is None:
-            queryset = queryset.filter(user=user, archive=0)
+            current_user = self.request.user.id
+            queryset1 = queryset.filter(user=user, archive=0)
+            queryset2 = queryset.filter(sharedwith=current_user)
```

Comment:
I fear this is erroneous.  `current_user = self.request.user.id` is the current user id not the user object. Kindly elaborate if that's not the case.